renku_notebooks/api/schemas/servers_patch.py

```diff
@@ -17,4 +17,5 @@ def list(cls):
 class PatchServerRequest(Schema):
     """Simple Enum for server status."""
 
-    state = fields.String(required=True, validate=validate.OneOf(PatchServerStatusEnum.list()))
+    state = fields.String(required=False, validate=validate.OneOf(PatchServerStatusEnum.list()))
+    resource_class_id = fields.Int(required=False, validate=lambda x: x > 0)
```

Comment:
The `resource_class_id` is not part of the Session object given by the notebooks API. Would it be possible to store it somewhere (annotations?) so that the UI can know which is the current resource class?

Reply:
Ok so the resource class is now in the annotations like `"renku.io/resourceClassId": "2"`.

However keep in mind that this field may be completely absent in the response because we never recorded the resource class in the JupyterServer resource up until this PR.

So the UI should handle this. I have no way to figure out what resource class was used to start the session on the sessions that are already there or launched prior to this PR.

Reply:
👍